### DISCUSSION TITLE
Why on_event('shutdown') in function not work

### DISCUSSION THREAD

**Discussion by dongfengweixiao** [fastapi/fastapi#6391] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

def register_init(app: FastAPI) -> None:
    @app.on_event("startup")
    async def init_connect():
        print("init_connect...")

    @app.on_event('shutdown')
    async def shutdown_connect():
        print("shutdown_connect...")


def create_app() -> FastAPI:
    app = FastAPI()
    
    register_init(app) # comment this line and uncomment code block blow
    
    return app


app = create_app()


# @app.on_event("startup")
# async def init_connect():
#     print("init_connect...")


# @app.on_event('shutdown')
# async def shutdown_connect():
#     print("shutdown_connect...")


if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app='main:app', host="127.0.0.1", port=8010, reload=True, debug=True)
```

### Description

Default code only print "init_connect"

uncomment can print "init_connect" and "shutdown_connect".

### Operating System

Windows

### FastAPI Version

0.75.2

### Python Version

Python 3.10.4

**Comment by Kludex** [comment]
Both work here... On Ubuntu with Python 3.8.12.

Not sure what would differ on that sense... 🤔

**Comment by Kludex** [comment, marked as answer]
There's a bug on uvicorn when using Windows.

This is going to be solved on: https://github.com/encode/uvicorn/pull/1584.
